# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/2d/HelpersSkia.h
# Commit: 20b46afb225f
# Full Hash: 20b46afb225f7b0076e3fc0c000ce609ac79e0ba
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2023-04-11 09:27:51
# Regressor Bug: 1821512
# File Overlap Count: 1
# Description:
#   Bug 1821512 - Fix Skia usage for API changes. r=jrmuizel
#   
#   This just tries to address fairly random changes in the Skia API and correct
#   our usage of it in Moz2D and some other places.
#   
# ==============================================================================

diff -r b531f604ea4c -r 20b46afb225f gfx/2d/HelpersSkia.h
--- a/gfx/2d/HelpersSkia.h	Tue Apr 11 05:00:04 2023 +0000
+++ b/gfx/2d/HelpersSkia.h	Tue Apr 11 05:00:05 2023 +0000
@@ -9,8 +9,10 @@
 
 #include "2D.h"
 #include "skia/include/core/SkCanvas.h"
+#include "skia/include/core/SkPathEffect.h"
+#include "skia/include/core/SkPathTypes.h"
+#include "skia/include/core/SkShader.h"
 #include "skia/include/effects/SkDashPathEffect.h"
-#include "skia/include/core/SkShader.h"
 #include "mozilla/Assertions.h"
 #include <cmath>
 #include <vector>
@@ -146,8 +148,8 @@
           SkFloatToScalar(aOptions.mDashPattern[i % aOptions.mDashLength]);
     }
 
-    sk_sp<SkPathEffect> dash = SkDashPathEffect::Make(
-        &pattern.front(), dashCount, SkFloatToScalar(aOptions.mDashOffset));
+    auto dash = SkDashPathEffect::Make(&pattern.front(), dashCount,
+                                       SkFloatToScalar(aOptions.mDashOffset));
     aPaint.setPathEffect(dash);
   }
 
@@ -303,14 +305,14 @@
   return SkFontHinting::kNormal;
 }
 
-static inline FillRule GetFillRule(SkPath::FillType aFillType) {
+static inline FillRule GetFillRule(SkPathFillType aFillType) {
   switch (aFillType) {
-    case SkPath::kWinding_FillType:
+    case SkPathFillType::kWinding:
       return FillRule::FILL_WINDING;
-    case SkPath::kEvenOdd_FillType:
+    case SkPathFillType::kEvenOdd:
       return FillRule::FILL_EVEN_ODD;
-    case SkPath::kInverseWinding_FillType:
-    case SkPath::kInverseEvenOdd_FillType:
+    case SkPathFillType::kInverseWinding:
+    case SkPathFillType::kInverseEvenOdd:
     default:
       NS_WARNING("Unsupported fill type\n");
       break;